In [31]:
%matplotlib inline
#Librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statistics as stat
from scipy import stats as st
import scipy.stats as st
#Comprobar
from sklearn.model_selection import train_test_split as tts
from sklearn.metrics import mean_squared_error as mse
from sklearn.model_selection import RepeatedStratifiedKFold
from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import cross_val_score as cvs
from sklearn.model_selection import RandomizedSearchCV
from sklearn.metrics import accuracy_score
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
#Modificar
from sklearn.preprocessing import LabelEncoder
#Modelos Regresion
from sklearn.ensemble import RandomForestRegressor as RFR  
from sklearn.tree import ExtraTreeRegressor as ETR
from sklearn.tree import DecisionTreeRegressor as DTR
from sklearn.ensemble import GradientBoostingRegressor as GBR
from xgboost import XGBRegressor as XGBR
from catboost import CatBoostRegressor as CTR
from lightgbm import LGBMRegressor as LGBMR
from xgboost import XGBRFRegressor as XGBRF
from sklearn.linear_model import LinearRegression as LinReg
from sklearn.neighbors import KNeighborsRegressor as KNR
from sklearn.linear_model import LogisticRegression as LogReg
#Modelos Clasificadores
from xgboost import XGBClassifier
from sklearn.ensemble import RandomForestClassifier
from catboost import CatBoostClassifier
from sklearn.ensemble import BaggingClassifier
#Otros
import warnings
warnings.filterwarnings("ignore")
pd.set_option('display.max_columns', None)
#plt.rcParams['figure.figsize']=(20, 20)

In [32]:
data = pd.read_csv('../final_match.csv')
data.head()

Id  Div        Date  HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG  \
0   0  SP2  2022-12-08  UD Ibiza     Racing Sant     1     0   H     0     0   
1   1  SP2  2022-12-08    Burgos           Eibar     1     2   A     1     0   
2   2  SP2  2022-12-08    Malaga         Granada     1     1   D     1     0   
3   3  SP2  2022-12-08   Levante    Ponferradina     0     0   D     0     0   
4   4  SP2  2022-12-08      Lugo  Sporting Gijon     0     1   A     0     0   

  HTR  HS  AS  HST  AST  HF  AF  HC  AC  HY  AY  HR  AR  B365H  B365D  B365A  \
0   D   6   7    4    4  22  13   3   6   1   4   0   1    2.7   3.10   2.75   
1   H   7  17    3    5   4  12   2   7   4   2   0   0    3.1   3.00   2.50   
2   H  11  16    4    8  15  13   5   4   4   2   0   0    3.1   3.25   2.37   
3   D  19   3    7    0   6  12   7   6   4   2   0   0    1.5   4.20   7.00   
4   D  14   9    1    2  10  15   7   2   2   4   0   1    3.4   3.10   2.30   

    BWH  BWD   BWA  Season  Id_HomeTeam  Id_AwayTeam  HTGS  ATGS  HTGC  ATGC  \
0  2.55  2.9  2.75    2223           39           36    13     9    25    13   
1  3.00  2.8  2.45    2223           23           26    17    21     9    16   
2  3.00  3.0  2.30    2223           33           27    14    21    23    14   
3  1.44  4.1  6.75    2223           31           35    20    15    13    25   
4  3.10  2.9  2.30    2223           32           37    16    21    23    20   

   HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  \
0   15   18   L   L   D   L   L   L   W   W   W   D        LDLWW        LLLWD   
1   31   30   D   D   D   W   W   L   D   W   D   L        DDWDD        DWLWL   
2   15   26   L   W   W   W   L   L   L   L   L   W        LWLLL        WWLLW   
3   31   15   D   W   W   L   D   L   W   L   L   W        DWDWL        WLLLW   
4   18   25   D   W   W   D   W   W   L   L   L   L        DWWLL        WDWLL   

   HTFormPts  ATFormPts  HTWin3  HTWin5  HTLoss3  HTLoss5  ATWin3  ATWin5  \
0          7          4       0       0        0        0       0       0   
1          7          7       0       0        0        0       0       0   
2          3          9       0       0        1        0       0       0   
3          8          6       0       0        0        0       0       0   
4          7          7       0       0        0        0       0       0   

   ATLoss3  ATLoss5  HTDG  ATDG  DiffPts  DiffFormPts  Journey  Med_HTDG  \
0        0        0   -12    -4       -3            3       19 -0.631579   
1        0        0     8     5        1            0       19  0.421053   
2        0        0    -9     7      -11           -6       19 -0.473684   
3        0        0     7   -10       16            2       19  0.368421   
4        0        0    -7     1       -7            0       19 -0.368421   

   Med_ATDG  Med_DiffPts  Med_DiffFormPts   Med_HTP   Med_ATP  Med_HTGS  \
0 -0.210526    -0.157895         0.157895  0.789474  0.947368  0.684211   
1  0.263158     0.052632         0.000000  1.631579  1.578947  0.894737   
2  0.368421    -0.578947        -0.315789  0.789474  1.368421  0.736842   
3 -0.526316     0.842105         0.105263  1.631579  0.789474  1.052632   
4  0.052632    -0.368421         0.000000  0.947368  1.315789  0.842105   

   Med_ATGS  Med_HTGC  Med_ATGC FTR_h_or_not  
0  0.473684  1.315789  0.684211            H  
1  1.105263  0.473684  0.842105           NH  
2  1.105263  1.210526  0.736842           NH  
3  0.789474  0.684211  1.315789           NH  
4  1.105263  1.210526  1.052632           NH

In [33]:
cat_cols = data.select_dtypes(include = ['object','category']).columns
num_cols = data.select_dtypes(exclude = ['object','category']).columns

cat_cols, num_cols

(Index(['Div', 'Date', 'HomeTeam', 'AwayTeam', 'FTR', 'HTR', 'HM1', 'AM1',
        'HM2', 'AM2', 'HM3', 'AM3', 'HM4', 'AM4', 'HM5', 'AM5', 'HTFormPtsStr',
        'ATFormPtsStr', 'FTR_h_or_not'],
       dtype='object'),
 Index(['Id', 'FTHG', 'FTAG', 'HTHG', 'HTAG', 'HS', 'AS', 'HST', 'AST', 'HF',
        'AF', 'HC', 'AC', 'HY', 'AY', 'HR', 'AR', 'B365H', 'B365D', 'B365A',
        'BWH', 'BWD', 'BWA', 'Season', 'Id_HomeTeam', 'Id_AwayTeam', 'HTGS',
        'ATGS', 'HTGC', 'ATGC', 'HTP', 'ATP', 'HTFormPts', 'ATFormPts',
        'HTWin3', 'HTWin5', 'HTLoss3', 'HTLoss5', 'ATWin3', 'ATWin5', 'ATLoss3',
        'ATLoss5', 'HTDG', 'ATDG', 'DiffPts', 'DiffFormPts', 'Journey',
        'Med_HTDG', 'Med_ATDG', 'Med_DiffPts', 'Med_DiffFormPts', 'Med_HTP',
        'Med_ATP', 'Med_HTGS', 'Med_ATGS', 'Med_HTGC', 'Med_ATGC'],
       dtype='object'))

In [34]:
data.info(memory_usage = 'deep')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14663 entries, 0 to 14662
Data columns (total 76 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Id               14663 non-null  int64  
 1   Div              14663 non-null  object 
 2   Date             14663 non-null  object 
 3   HomeTeam         14663 non-null  object 
 4   AwayTeam         14663 non-null  object 
 5   FTHG             14663 non-null  int64  
 6   FTAG             14663 non-null  int64  
 7   FTR              14663 non-null  object 
 8   HTHG             14663 non-null  int64  
 9   HTAG             14663 non-null  int64  
 10  HTR              14663 non-null  object 
 11  HS               14663 non-null  int64  
 12  AS               14663 non-null  int64  
 13  HST              14663 non-null  int64  
 14  AST              14663 non-null  int64  
 15  HF               14663 non-null  int64  
 16  AF               14663 non-null  int64  
 17  HC          

In [35]:
day = []
month = []
year = []
for x in data.Date:
    x = x.split('-')
    day.append(int(x[2]))
    month.append(int(x[1]))
    year.append(int(x[0]))
data['year'] = year
data['month'] = month
data['day'] = day

data = data.drop(['Date'], axis = 1)
data.head()

Id  Div  HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  HS  AS  \
0   0  SP2  UD Ibiza     Racing Sant     1     0   H     0     0   D   6   7   
1   1  SP2    Burgos           Eibar     1     2   A     1     0   H   7  17   
2   2  SP2    Malaga         Granada     1     1   D     1     0   H  11  16   
3   3  SP2   Levante    Ponferradina     0     0   D     0     0   D  19   3   
4   4  SP2      Lugo  Sporting Gijon     0     1   A     0     0   D  14   9   

   HST  AST  HF  AF  HC  AC  HY  AY  HR  AR  B365H  B365D  B365A   BWH  BWD  \
0    4    4  22  13   3   6   1   4   0   1    2.7   3.10   2.75  2.55  2.9   
1    3    5   4  12   2   7   4   2   0   0    3.1   3.00   2.50  3.00  2.8   
2    4    8  15  13   5   4   4   2   0   0    3.1   3.25   2.37  3.00  3.0   
3    7    0   6  12   7   6   4   2   0   0    1.5   4.20   7.00  1.44  4.1   
4    1    2  10  15   7   2   2   4   0   1    3.4   3.10   2.30  3.10  2.9   

    BWA  Season  Id_HomeTeam  Id_AwayTeam  HTGS  ATGS  HTGC  ATGC  HTP  ATP  \
0  2.75    2223           39           36    13     9    25    13   15   18   
1  2.45    2223           23           26    17    21     9    16   31   30   
2  2.30    2223           33           27    14    21    23    14   15   26   
3  6.75    2223           31           35    20    15    13    25   31   15   
4  2.30    2223           32           37    16    21    23    20   18   25   

  HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  \
0   L   L   D   L   L   L   W   W   W   D        LDLWW        LLLWD   
1   D   D   D   W   W   L   D   W   D   L        DDWDD        DWLWL   
2   L   W   W   W   L   L   L   L   L   W        LWLLL        WWLLW   
3   D   W   W   L   D   L   W   L   L   W        DWDWL        WLLLW   
4   D   W   W   D   W   W   L   L   L   L        DWWLL        WDWLL   

   HTFormPts  ATFormPts  HTWin3  HTWin5  HTLoss3  HTLoss5  ATWin3  ATWin5  \
0          7          4       0       0        0        0       0       0   
1          7          7       0       0        0        0       0       0   
2          3          9       0       0        1        0       0       0   
3          8          6       0       0        0        0       0       0   
4          7          7       0       0        0        0       0       0   

   ATLoss3  ATLoss5  HTDG  ATDG  DiffPts  DiffFormPts  Journey  Med_HTDG  \
0        0        0   -12    -4       -3            3       19 -0.631579   
1        0        0     8     5        1            0       19  0.421053   
2        0        0    -9     7      -11           -6       19 -0.473684   
3        0        0     7   -10       16            2       19  0.368421   
4        0        0    -7     1       -7            0       19 -0.368421   

   Med_ATDG  Med_DiffPts  Med_DiffFormPts   Med_HTP   Med_ATP  Med_HTGS  \
0 -0.210526    -0.157895         0.157895  0.789474  0.947368  0.684211   
1  0.263158     0.052632         0.000000  1.631579  1.578947  0.894737   
2  0.368421    -0.578947        -0.315789  0.789474  1.368421  0.736842   
3 -0.526316     0.842105         0.105263  1.631579  0.789474  1.052632   
4  0.052632    -0.368421         0.000000  0.947368  1.315789  0.842105   

   Med_ATGS  Med_HTGC  Med_ATGC FTR_h_or_not  year  month  day  
0  0.473684  1.315789  0.684211            H  2022     12    8  
1  1.105263  0.473684  0.842105           NH  2022     12    8  
2  1.105263  1.210526  0.736842           NH  2022     12    8  
3  0.789474  0.684211  1.315789           NH  2022     12    8  
4  1.105263  1.210526  1.052632           NH  2022     12    8

In [36]:
def diction(col):
    lista = list(data[col].unique())
    nums =[i for i in range(len(lista))]
    print({(lista[i]): (nums[i]) for i in range(len(lista))})
    return {(lista[i]): (nums[i]) for i in range(len(lista))}
data.select_dtypes(include = ['object','category'])

Div         HomeTeam        AwayTeam FTR HTR HM1 AM1 HM2 AM2 HM3 AM3  \
0      SP2         UD Ibiza     Racing Sant   H   D   L   L   D   L   L   L   
1      SP2           Burgos           Eibar   A   H   D   D   D   W   W   L   
2      SP2           Malaga         Granada   D   H   L   W   W   W   L   L   
3      SP2          Levante    Ponferradina   D   D   D   W   W   L   D   L   
4      SP2             Lugo  Sporting Gijon   A   D   D   W   W   D   W   W   
...    ...              ...             ...  ..  ..  ..  ..  ..  ..  ..  ..   
14658  SP1            Cadiz     Real Madrid   A   A   M   M   M   M   M   M   
14659  SP1  Atletico Madrid        Zaragoza   D   D   M   M   M   M   M   M   
14660  SP1           Alaves       Barcelona   D   D   M   M   M   M   M   M   
14661  SP1         Valencia           Betis   H   D   M   M   M   M   M   M   
14662  SP1    Athletic Club   Real Sociedad   H   D   M   M   M   M   M   M   

      HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr FTR_h_or_not  
0       W   W   W   D        LDLWW        LLLWD            H  
1       D   W   D   L        DDWDD        DWLWL           NH  
2       L   L   L   W        LWLLL        WWLLW           NH  
3       W   L   L   W        DWDWL        WLLLW           NH  
4       L   L   L   L        DWWLL        WDWLL           NH  
...    ..  ..  ..  ..          ...          ...          ...  
14658   M   M   M   M        MMMMM        MMMMM           NH  
14659   M   M   M   M        MMMMM        MMMMM           NH  
14660   M   M   M   M        MMMMM        MMMMM           NH  
14661   M   M   M   M        MMMMM        MMMMM            H  
14662   M   M   M   M        MMMMM        MMMMM            H  

[14663 rows x 18 columns]

In [37]:
form_dict ={
    'D': 0,
    'L': 1,
    'W': 2,
    'M': 3
}
res_dict ={
    'D': 0,
    'A': 1,
    'H': 2
}

In [38]:
data_endode = data.copy()
#Creación de diccionarios
dictio_div = diction('Div')
dictio_hometeam = diction('HomeTeam')
dictio_HTFormPtsStr = diction('HTFormPtsStr')
dictio_ATFormPtsStr = diction('ATFormPtsStr')
dictio_HTFormPtsStr.update(dictio_ATFormPtsStr)
dictio_FTR_h_or_not = diction('FTR_h_or_not')
# Transformación
data_endode.Div = data_endode.Div.apply(lambda x: dictio_div[x])
data_endode.HomeTeam = data_endode.HomeTeam.apply(lambda x: dictio_hometeam[x])
data_endode.AwayTeam = data_endode.AwayTeam.apply(lambda x: dictio_hometeam[x])
data_endode[['FTR','HTR']] = data_endode[['FTR','HTR']].replace(res_dict)
data_endode.HTFormPtsStr = data_endode.HTFormPtsStr.apply(lambda x: dictio_HTFormPtsStr[x])
data_endode.ATFormPtsStr = data_endode.ATFormPtsStr.apply(lambda x: dictio_HTFormPtsStr[x])
data_endode.FTR_h_or_not = data_endode.FTR_h_or_not.apply(lambda x: dictio_FTR_h_or_not[x])
data_endode[['HM1','AM1','HM2','AM2','HM3','AM3','HM4','AM4','HM5','AM5']] = data_endode[['HM1','AM1','HM2','AM2','HM3','AM3','HM4','AM4','HM5','AM5']].replace(form_dict)
data_endode.head()

{'SP2': 0, 'SP1': 1}
{'UD Ibiza': 0, 'Burgos': 1, 'Malaga': 2, 'Levante': 3, 'Lugo': 4, 'Tenerife': 5, 'Cartagena': 6, 'Huesca': 7, 'Albacete': 8, 'Oviedo': 9, 'Leganes': 10, 'Ponferradina': 11, 'Eibar': 12, 'Racing Sant': 13, 'Andorra': 14, 'CD Mirandes': 15, 'Zaragoza': 16, 'Villarreal B': 17, 'Sporting Gijon': 18, 'Granada': 19, 'Las Palmas': 20, 'Alaves': 21, 'Valladolid': 22, 'Alcorcon': 23, 'Amorebieta': 24, 'Real Sociedad B': 25, 'Almeria': 26, 'Girona': 27, 'Fuenlabrada': 28, 'Logrones': 29, 'Rayo Vallecano': 30, 'Castellon': 31, 'Sabadell': 32, 'Espanyol': 33, 'Mallorca': 34, 'La Coruna': 35, 'Elche': 36, 'Cadiz': 37, 'Numancia': 38, 'Extremadura UD': 39, 'Gimnastic': 40, 'Rayo Majadahonda': 41, 'Osasuna': 42, 'Reus Deportiu': 43, 'Cordoba': 44, 'Barcelona B': 45, 'Lorca': 46, 'Leonesa': 47, 'Sevilla B': 48, 'Getafe': 49, 'UCAM Murcia': 50, 'Athletic Club B': 51, 'Llagostera': 52, 'Betis': 53, 'Recreativo': 54, 'Hercules': 55, 'Jaen': 56, 'Murcia': 57, 'Real Madrid B': 58, 'Xe

Id  Div  HomeTeam  AwayTeam  FTHG  FTAG  FTR  HTHG  HTAG  HTR  HS  AS  HST  \
0   0    0         0        13     1     0    2     0     0    0   6   7    4   
1   1    0         1        12     1     2    1     1     0    2   7  17    3   
2   2    0         2        19     1     1    0     1     0    2  11  16    4   
3   3    0         3        11     0     0    0     0     0    0  19   3    7   
4   4    0         4        18     0     1    1     0     0    0  14   9    1   

   AST  HF  AF  HC  AC  HY  AY  HR  AR  B365H  B365D  B365A   BWH  BWD   BWA  \
0    4  22  13   3   6   1   4   0   1    2.7   3.10   2.75  2.55  2.9  2.75   
1    5   4  12   2   7   4   2   0   0    3.1   3.00   2.50  3.00  2.8  2.45   
2    8  15  13   5   4   4   2   0   0    3.1   3.25   2.37  3.00  3.0  2.30   
3    0   6  12   7   6   4   2   0   0    1.5   4.20   7.00  1.44  4.1  6.75   
4    2  10  15   7   2   2   4   0   1    3.4   3.10   2.30  3.10  2.9  2.30   

   Season  Id_HomeTeam  Id_AwayTeam  HTGS  ATGS  HTGC  ATGC  HTP  ATP  HM1  \
0    2223           39           36    13     9    25    13   15   18    1   
1    2223           23           26    17    21     9    16   31   30    0   
2    2223           33           27    14    21    23    14   15   26    1   
3    2223           31           35    20    15    13    25   31   15    0   
4    2223           32           37    16    21    23    20   18   25    0   

   AM1  HM2  AM2  HM3  AM3  HM4  AM4  HM5  AM5  HTFormPtsStr  ATFormPtsStr  \
0    1    0    1    1    1    2    2    2    0           146             0   
1    0    0    2    2    1    0    2    0    1            68             1   
2    2    2    2    1    1    1    1    1    2           185             2   
3    2    2    1    0    1    2    1    1    2           101             3   
4    2    2    0    2    2    1    1    1    1           148             4   

   HTFormPts  ATFormPts  HTWin3  HTWin5  HTLoss3  HTLoss5  ATWin3  ATWin5  \
0          7          4       0       0        0        0       0       0   
1          7          7       0       0        0        0       0       0   
2          3          9       0       0        1        0       0       0   
3          8          6       0       0        0        0       0       0   
4          7          7       0       0        0        0       0       0   

   ATLoss3  ATLoss5  HTDG  ATDG  DiffPts  DiffFormPts  Journey  Med_HTDG  \
0        0        0   -12    -4       -3            3       19 -0.631579   
1        0        0     8     5        1            0       19  0.421053   
2        0        0    -9     7      -11           -6       19 -0.473684   
3        0        0     7   -10       16            2       19  0.368421   
4        0        0    -7     1       -7            0       19 -0.368421   

   Med_ATDG  Med_DiffPts  Med_DiffFormPts   Med_HTP   Med_ATP  Med_HTGS  \
0 -0.210526    -0.157895         0.157895  0.789474  0.947368  0.684211   
1  0.263158     0.052632         0.000000  1.631579  1.578947  0.894737   
2  0.368421    -0.578947        -0.315789  0.789474  1.368421  0.736842   
3 -0.526316     0.842105         0.105263  1.631579  0.789474  1.052632   
4  0.052632    -0.368421         0.000000  0.947368  1.315789  0.842105   

   Med_ATGS  Med_HTGC  Med_ATGC  FTR_h_or_not  year  month  day  
0  0.473684  1.315789  0.684211             0  2022     12    8  
1  1.105263  0.473684  0.842105             1  2022     12    8  
2  1.105263  1.210526  0.736842             1  2022     12    8  
3  0.789474  0.684211  1.315789             1  2022     12    8  
4  1.105263  1.210526  1.052632             1  2022     12    8

In [39]:
def mean_n(df):
    return df.apply(lambda x: (x-x.mean())/ x.std(), axis = 0)

In [40]:
X = data_endode.drop(['FTR'], axis = 1)[['Id', 'Div', 'HomeTeam', 'AwayTeam', 'B365H', 'B365D', 'B365A', 'BWH',
       'BWD', 'BWA', 'Season', 'HTGS', 'ATGS', 'HTGC', 'ATGC', 'HTP', 'ATP',
       'HM1', 'AM1', 'HM2', 'AM2', 'HM3', 'AM3', 'HM4', 'AM4', 'HM5', 'AM5',
       'HTFormPtsStr', 'ATFormPtsStr', 'HTFormPts', 'ATFormPts', 'HTWin3',
       'HTWin5', 'HTLoss3', 'HTLoss5', 'ATLoss3', 'HTDG', 'ATDG', 'DiffPts',
       'DiffFormPts', 'Journey', 'Med_HTDG', 'Med_ATDG', 'Med_DiffPts',
       'Med_DiffFormPts', 'Med_HTP', 'Med_ATP', 'Med_HTGS', 'Med_ATGS',
       'Med_HTGC', 'Med_ATGC', 'year', 'month', 'day']]
y = data_endode.FTR
#print(data_endode.FTR)
X = mean_n(X)
#print(X)
X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2, random_state=0, stratify = y)

X_train.shape, X_test.shape, y_train.shape, y_test.shape

((11730, 54), (2933, 54), (11730,), (2933,))

In [11]:
model_fin = CatBoostClassifier(random_state = 42, silent = True, thread_count = 1, iterations = 1100, learning_rate = 0.02, 
                           max_depth = 5).fit(X_train, y_train)

y_pred = model_fin.predict(X_test)
y_pred[:10]

array([[0],
       [0],
       [2],
       [2],
       [2],
       [2],
       [1],
       [1],
       [2],
       [2]], dtype=int64)

In [12]:
accuracy_score(y_pred, y_test)

0.5069894306171155

In [41]:
data[['Id', 'Div', 'HomeTeam', 'AwayTeam', 'B365H', 'B365D', 'B365A', 'BWH',
       'BWD', 'BWA', 'Season', 'HTGS', 'ATGS', 'HTGC', 'ATGC', 'HTP', 'ATP',
       'HM1', 'AM1', 'HM2', 'AM2', 'HM3', 'AM3', 'HM4', 'AM4', 'HM5', 'AM5',
       'HTFormPtsStr', 'ATFormPtsStr', 'HTFormPts', 'ATFormPts', 'HTWin3',
       'HTWin5', 'HTLoss3', 'HTLoss5', 'ATLoss3', 'HTDG', 'ATDG', 'DiffPts',
       'DiffFormPts', 'Journey', 'Med_HTDG', 'Med_ATDG', 'Med_DiffPts',
       'Med_DiffFormPts', 'Med_HTP', 'Med_ATP', 'Med_HTGS', 'Med_ATGS',
       'Med_HTGC', 'Med_ATGC', 'year', 'month', 'day']].head(11)

Id  Div   HomeTeam        AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0    0  SP2   UD Ibiza     Racing Sant   2.70   3.10   2.75  2.55  2.90  2.75   
1    1  SP2     Burgos           Eibar   3.10   3.00   2.50  3.00  2.80  2.45   
2    2  SP2     Malaga         Granada   3.10   3.25   2.37  3.00  3.00  2.30   
3    3  SP2    Levante    Ponferradina   1.50   4.20   7.00  1.44  4.10  6.75   
4    4  SP2       Lugo  Sporting Gijon   3.40   3.10   2.30  3.10  2.90  2.30   
5    5  SP2   Tenerife          Alaves   2.37   2.90   3.50  2.30  2.75  3.30   
6    6  SP2  Cartagena    Villarreal B   1.72   3.60   5.25  1.73  3.40  4.50   
7    7  SP2     Huesca         Andorra   2.62   3.00   3.10  2.55  2.80  2.85   
8    8  SP2   Albacete        Zaragoza   2.50   3.00   3.20  2.45  2.95  2.80   
9    9  SP2     Oviedo      Las Palmas   3.00   2.90   2.55  3.00  2.85  2.40   
10  10  SP2    Leganes     CD Mirandes   2.00   3.10   4.50  1.95  3.00  3.90   

    Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0     2223    13     9    25    13   15   18   L   L   D   L   L   L   W   W   
1     2223    17    21     9    16   31   30   D   D   D   W   W   L   D   W   
2     2223    14    21    23    14   15   26   L   W   W   W   L   L   L   L   
3     2223    20    15    13    25   31   15   D   W   W   L   D   L   W   L   
4     2223    16    21    23    20   18   25   D   W   W   D   W   W   L   L   
5     2223    18    20    19    16   23   31   D   W   L   D   W   D   L   W   
6     2223    21    19    15    22   31   24   W   D   W   L   L   W   W   L   
7     2223    16    19    14    15   26   24   L   L   L   L   W   W   W   D   
8     2223    19    14    23    17   23   21   D   D   W   L   W   L   L   W   
9     2223    12    22    13    10   24   34   W   W   W   W   D   D   D   W   
10    2223    18    20    16    24   26   19   L   L   L   L   W   L   L   D   

   HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0    W   D        LDLWW        LLLWD          7          4       0       0   
1    D   L        DDWDD        DWLWL          7          7       0       0   
2    L   W        LWLLL        WWLLW          3          9       0       0   
3    L   W        DWDWL        WLLLW          8          6       0       0   
4    L   L        DWWLL        WDWLL          7          7       0       0   
5    W   W        DLWLW        WDDWW          7         11       0       0   
6    W   W        WWLWW        DLWLW         12          7       0       0   
7    L   W        LLWWL        LLWDW          6          7       0       0   
8    L   W        DWWLL        DLLWW          7          7       0       0   
9    L   D        WWDDL        WWDWD          8         11       0       0   
10   D   L        LLWLD        LLLDL          4          1       0       0   

    HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0         0        0        0   -12    -4       -3            3       19   
1         0        0        0     8     5        1            0       19   
2         1        0        0    -9     7      -11           -6       19   
3         0        0        0     7   -10       16            2       19   
4         0        0        0    -7     1       -7            0       19   
5         0        0        0    -1     4       -8           -4       19   
6         0        0        0     6    -3        7            5       19   
7         0        0        0     2     4        2           -1       19   
8         0        0        0    -4    -3        2            0       19   
9         0        0        0    -1    12      -10           -3       19   
10        0        0        0     2    -4        7            3       19   

    Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts   Med_HTP   Med_ATP  \
0  -0.631579 -0.210526    -0.157895         0.157895  0.789474  0.947368   
1   0.421053  0.263158     0.052632         0.000000  1.631579  1.57

In [42]:
dict_test = {
    'Id': [0,1,2,3,4,5,6,7,8,9,10],
    'Div': ['SP2','SP2','SP2','SP2','SP2','SP2','SP2','SP2','SP2','SP2','SP2'],
    'HomeTeam': ['Alaves','Las Palmas', 'Racing Sant', 'Granada', 'Eibar', 'Ponferradina', 'Sporting Gijon', 'UD Ibiza', 'Zaragoza', 'Andorra', 'Villarreal B'],
    'AwayTeam': ['Levante', 'Albacete', 'CD Mirandes', 'Burgos', 'Oviedo', 'Lugo', 'Cartagena', 'Malaga', 'Huesca', 'Leganes', 'Tenerife'],
    'B365H': [2.37,1.80,2.10,1.80,1.75,2.25,2.20,2.30,1.95,2.10,2.90],
    'B365D': [3.00,3.30,3.10,3.30,3.40,3.30,3.10,3.00,3.00,3.10,3.00],
    'B365A': [3.25,5.00,4.00,5.00,4.75,3.25,3.50,3.40,4.75,3.80,2.62],
    'BWH': [2.40,1.75,2.10,1.80,1.75,2.20,2.10,2.25,1.95,2.05,2.90],
    'BWD': [2.95,3.25,2.95,3.20,3.20,3.20,3.10,2.95,3.00,3.95,2.87],
    'BWA': [2.90,4.60,3.50,4.40,4.75,3.00,3.40,3.20,3.90,3.60,2.45],
    'Season': [2223,2223,2223,2223,2223,2223,2223,2223,2223,2223,2223],
    'HTGS': [22,22,9,23,23,18,22,13,14,20,21],
    'ATGS': [20,21,21,18,12,17,23,14,16,19,19],
    'HTGC': [17,10,15,14,17,27,21,27,17,15,22],
    'ATGC': [13,24,25,9,14,25,18,24,14,18,21],
    'HTP': [34,35,18,29,33,18,26,15,22,27,27],
    'ATP': [32,26,20,34,24,18,31,15,27,26,23],
    'HM1': ['W','D','W','L','D','L','D','D','L','L','L'],
    'AM1': ['W','L','D','L','W','D','D','D','W','W','W'],
    'HM2': ['D','D','L','W','D','D','D','L','D','W','W'],
    'AM2': ['D','L','W','W','D','L','L','D','L','D','W'],
    'HM3': ['W','W','L','L','L','L','D','L','D','W','L'],
    'AM3': ['D','W','L','D','W','D','W','W','D','W','L'],
    'HM4': ['L','W','L','W','W','L','L','L','W','L','D'],
    'AM4': ['W','L','W','W','W','W','W','L','L','D','L'],
    'HM5': ['L','D','L','D','W','D','W','W','D','L','W'],
    'AM5': ['D','D','D','L','D','L','L','D','W','D','W'],
    'HTFormPtsStr': ['WDWLL','DDWWD','WLLLL','LWLWD','DDLWW','LDLLD','DDDLW','DLLLW','LDDWD','LWWLL','LWLDW'],
    'ATFormPtsStr': ['WDDWD','LLWLD','DWLWD','LWDWL','WDWWD','DLDWL','DLWWL','DDWLD','WLDLW','WDWDD','WWLLW'],
    'HTFormPts': [7,9,3,7,8,2,6,4,6,6,7],
    'ATFormPts': [9,4,8,7,11,5,7,6,7,9,9]
}
testeo = pd.DataFrame(dict_test)
testeo

Id  Div        HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD  \
0    0  SP2          Alaves      Levante   2.37    3.0   3.25  2.40  2.95   
1    1  SP2      Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25   
2    2  SP2     Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95   
3    3  SP2         Granada       Burgos   1.80    3.3   5.00  1.80  3.20   
4    4  SP2           Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20   
5    5  SP2    Ponferradina         Lugo   2.25    3.3   3.25  2.20  3.20   
6    6  SP2  Sporting Gijon    Cartagena   2.20    3.1   3.50  2.10  3.10   
7    7  SP2        UD Ibiza       Malaga   2.30    3.0   3.40  2.25  2.95   
8    8  SP2        Zaragoza       Huesca   1.95    3.0   4.75  1.95  3.00   
9    9  SP2         Andorra      Leganes   2.10    3.1   3.80  2.05  3.95   
10  10  SP2    Villarreal B     Tenerife   2.90    3.0   2.62  2.90  2.87   

     BWA  Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3  \
0   2.90    2223    22    20    17    13   34   32   W   W   D   D   W   D   
1   4.60    2223    22    21    10    24   35   26   D   L   D   L   W   W   
2   3.50    2223     9    21    15    25   18   20   W   D   L   W   L   L   
3   4.40    2223    23    18    14     9   29   34   L   L   W   W   L   D   
4   4.75    2223    23    12    17    14   33   24   D   W   D   D   L   W   
5   3.00    2223    18    17    27    25   18   18   L   D   D   L   L   D   
6   3.40    2223    22    23    21    18   26   31   D   D   D   L   D   W   
7   3.20    2223    13    14    27    24   15   15   D   D   L   D   L   W   
8   3.90    2223    14    16    17    14   22   27   L   W   D   L   D   D   
9   3.60    2223    20    19    15    18   27   26   L   W   W   D   W   W   
10  2.45    2223    21    19    22    21   27   23   L   W   W   W   L   L   

   HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  
0    L   W   L   D        WDWLL        WDDWD          7          9  
1    W   L   D   D        DDWWD        LLWLD          9          4  
2    L   W   L   D        WLLLL        DWLWD          3          8  
3    W   W   D   L        LWLWD        LWDWL          7          7  
4    W   W   W   D        DDLWW        WDWWD          8         11  
5    L   W   D   L        LDLLD        DLDWL          2          5  
6    L   W   W   L        DDDLW        DLWWL          6          7  
7    L   L   W   D        DLLLW        DDWLD          4          6  
8    W   L   D   W        LDDWD        WLDLW          6          7  
9    L   D   L   D        LWWLL        WDWDD          6          9  
10   D   L   W   W        LWLDW        WWLLW          7          9

In [43]:
def get_3w(s):
    if s[-3:] == 'WWW':
        return 1
    else:
        return 0
    
def get_5w(s):
    if s == 'WWWWW':
        return 1
    else:
        return 0
    
def get_3l(s):
    if s[-3:] == 'LLL':
        return 1
    else:
        return 0
    
def get_5l(s):
    if s == 'LLLLL':
        return 1
    else:
        return 0

In [44]:
testeo['HTWin3'] = testeo['HTFormPtsStr'].apply(get_3w)
testeo['HTWin5'] = testeo['HTFormPtsStr'].apply(get_5w)
testeo['HTLoss3'] = testeo['HTFormPtsStr'].apply(get_3l)
testeo['HTLoss5'] = testeo['HTFormPtsStr'].apply(get_5l)

testeo['ATLoss3'] = testeo['ATFormPtsStr'].apply(get_3l)

testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  
0        0        0        0  
1        0        0        0  
2        1        0        0  
3        0        0        0  
4        0        0        0

In [45]:
testeo['HTDG'] = testeo['HTGS'] - testeo['HTGC']
testeo['ATDG'] = testeo['ATGS'] - testeo['ATGC']
testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  
0        0        0        0     5     7  
1        0        0        0    12    -3  
2        1        0        0    -6    -4  
3        0        0        0     9     9  
4        0        0        0     6    -2

In [46]:
testeo['DiffPts'] = testeo['HTP'] - testeo['ATP']
testeo['DiffFormPts'] = testeo['HTFormPts'] - testeo['ATFormPts']

testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  
0        0        0        0     5     7        2           -2  
1        0        0        0    12    -3        9            5  
2        1        0        0    -6    -4       -2           -5  
3        0        0        0     9     9       -5            0  
4        0        0        0     6    -2        9           -3

In [47]:
testeo['Journey'] = 20
testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  
0        0        0        0     5     7        2           -2       20  
1        0        0        0    12    -3        9            5       20  
2        1        0        0    -6    -4       -2           -5       20  
3        0        0        0     9     9       -5            0       20  
4        0        0        0     6    -2        9           -3       20

In [48]:
testeo['Med_HTDG'] = testeo['HTDG'] / testeo.Journey
testeo['Med_ATDG'] = testeo['ATDG'] / testeo.Journey
testeo['Med_DiffPts'] = testeo['DiffPts'] / testeo.Journey
testeo['Med_DiffFormPts'] = testeo['DiffFormPts'] / testeo.Journey
testeo['Med_HTP'] = testeo['HTP'] / testeo.Journey
testeo['Med_ATP'] = testeo['ATP'] / testeo.Journey

testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0        0        0        0     5     7        2           -2       20   
1        0        0        0    12    -3        9            5       20   
2        1        0        0    -6    -4       -2           -5       20   
3        0        0        0     9     9       -5            0       20   
4        0        0        0     6    -2        9           -3       20   

   Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  Med_ATP  
0      0.25      0.35         0.10            -0.10     1.70      1.6  
1      0.60     -0.15         0.45             0.25     1.75      1.3  
2     -0.30     -0.20        -0.10            -0.25     0.90      1.0  
3      0.45      0.45        -0.25             0.00     1.45      1.7  
4      0.30     -0.10         0.45            -0.15     1.65      1.2

In [49]:
testeo['Med_HTGS'] = testeo['HTGS'] / testeo.Journey
testeo['Med_ATGS'] = testeo['ATGS'] / testeo.Journey
testeo['Med_HTGC'] = testeo['HTGC'] / testeo.Journey
testeo['Med_ATGC'] = testeo['ATGC'] / testeo.Journey
testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0        0        0        0     5     7        2           -2       20   
1        0        0        0    12    -3        9            5       20   
2        1        0        0    -6    -4       -2           -5       20   
3        0        0        0     9     9       -5            0       20   
4        0        0        0     6    -2        9           -3       20   

   Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  Med_ATP  \
0      0.25      0.35         0.10            -0.10     1.70      1.6   
1      0.60     -0.15         0.45             0.25     1.75      1.3   
2     -0.30     -0.20        -0.10            -0.25     0.90      1.0   
3      0.45      0.45        -0.25             0.00     1.45      1.7   
4      0.30     -0.10         0.45            -0.15     1.65      1.2   

   Med_HTGS  Med_ATGS  Med_HTGC  Med_ATGC  
0      1.10      1.00      0.85      0.65  
1      1.10      1.05      0.50      1.20  
2      0.45      1.05      0.75      1.25  
3      1.15      0.90      0.70      0.45  
4      1.15      0.60      0.85      0.70

In [50]:
testeo['year'] = 2022
testeo['month'] = 12
testeo['day'] = [12,11,11,11,11,11,11,11,10,10,10]
testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0        0        0        0     5     7        2           -2       20   
1        0        0        0    12    -3        9            5       20   
2        1        0        0    -6    -4       -2           -5       20   
3        0        0        0     9     9       -5            0       20   
4        0        0        0     6    -2        9           -3       20   

   Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  Med_ATP  \
0      0.25      0.35         0.10            -0.10     1.70      1.6   
1      0.60     -0.15         0.45             0.25     1.75      1.3   
2     -0.30     -0.20        -0.10            -0.25     0.90      1.0   
3      0.45      0.45        -0.25             0.00     1.45      1.7   
4      0.30     -0.10         0.45            -0.15     1.65      1.2   

   Med_HTGS  Med_ATGS  Med_HTGC  Med_ATGC  year  month  day  
0      1.10      1.00      0.85      0.65  2022     12   12  
1      1.10      1.05      0.50      1.20  2022     12   11  
2      0.45      1.05      0.75      1.25  2022     12   11  
3      1.15      0.90      0.70      0.45  2022     12   11  
4      1.15      0.60      0.85      0.70  2022     12   11

In [51]:
testeo.shape

(11, 54)

In [52]:
testeo.head()

Id  Div     HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  \
0   0  SP2       Alaves      Levante   2.37    3.0   3.25  2.40  2.95  2.90   
1   1  SP2   Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25  4.60   
2   2  SP2  Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95  3.50   
3   3  SP2      Granada       Burgos   1.80    3.3   5.00  1.80  3.20  4.40   
4   4  SP2        Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20  4.75   

   Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3 HM4 AM4  \
0    2223    22    20    17    13   34   32   W   W   D   D   W   D   L   W   
1    2223    22    21    10    24   35   26   D   L   D   L   W   W   W   L   
2    2223     9    21    15    25   18   20   W   D   L   W   L   L   L   W   
3    2223    23    18    14     9   29   34   L   L   W   W   L   D   W   W   
4    2223    23    12    17    14   33   24   D   W   D   D   L   W   W   W   

  HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0   L   D        WDWLL        WDDWD          7          9       0       0   
1   D   D        DDWWD        LLWLD          9          4       0       0   
2   L   D        WLLLL        DWLWD          3          8       0       0   
3   D   L        LWLWD        LWDWL          7          7       0       0   
4   W   D        DDLWW        WDWWD          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0        0        0        0     5     7        2           -2       20   
1        0        0        0    12    -3        9            5       20   
2        1        0        0    -6    -4       -2           -5       20   
3        0        0        0     9     9       -5            0       20   
4        0        0        0     6    -2        9           -3       20   

   Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  Med_ATP  \
0      0.25      0.35         0.10            -0.10     1.70      1.6   
1      0.60     -0.15         0.45             0.25     1.75      1.3   
2     -0.30     -0.20        -0.10            -0.25     0.90      1.0   
3      0.45      0.45        -0.25             0.00     1.45      1.7   
4      0.30     -0.10         0.45            -0.15     1.65      1.2   

   Med_HTGS  Med_ATGS  Med_HTGC  Med_ATGC  year  month  day  
0      1.10      1.00      0.85      0.65  2022     12   12  
1      1.10      1.05      0.50      1.20  2022     12   11  
2      0.45      1.05      0.75      1.25  2022     12   11  
3      1.15      0.90      0.70      0.45  2022     12   11  
4      1.15      0.60      0.85      0.70  2022     12   11

In [54]:
testeo_encode = testeo.copy()

In [55]:
testeo_encode.Div = testeo.Div.apply(lambda x: dictio_div[x])
testeo_encode.HomeTeam = testeo.HomeTeam.apply(lambda x: dictio_hometeam[x])
testeo_encode.AwayTeam = testeo.AwayTeam.apply(lambda x: dictio_hometeam[x])

testeo_encode.HTFormPtsStr = testeo.HTFormPtsStr.apply(lambda x: dictio_HTFormPtsStr[x])
testeo_encode.ATFormPtsStr = testeo.ATFormPtsStr.apply(lambda x: dictio_HTFormPtsStr[x])

testeo_encode[['HM1','AM1','HM2','AM2','HM3','AM3','HM4','AM4','HM5','AM5']] = testeo[['HM1','AM1','HM2','AM2','HM3','AM3','HM4','AM4','HM5','AM5']].replace(form_dict)
testeo_encode.head()

Id  Div  HomeTeam  AwayTeam  B365H  B365D  B365A   BWH   BWD   BWA  Season  \
0   0    0        21         3   2.37    3.0   3.25  2.40  2.95  2.90    2223   
1   1    0        20         8   1.80    3.3   5.00  1.75  3.25  4.60    2223   
2   2    0        13        15   2.10    3.1   4.00  2.10  2.95  3.50    2223   
3   3    0        19         1   1.80    3.3   5.00  1.80  3.20  4.40    2223   
4   4    0        12         9   1.75    3.4   4.75  1.75  3.20  4.75    2223   

   HTGS  ATGS  HTGC  ATGC  HTP  ATP  HM1  AM1  HM2  AM2  HM3  AM3  HM4  AM4  \
0    22    20    17    13   34   32    2    2    0    0    2    0    1    2   
1    22    21    10    24   35   26    0    1    0    1    2    2    2    1   
2     9    21    15    25   18   20    2    0    1    2    1    1    1    2   
3    23    18    14     9   29   34    1    1    2    2    1    0    2    2   
4    23    12    17    14   33   24    0    2    0    0    1    2    2    2   

   HM5  AM5  HTFormPtsStr  ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  HTWin5  \
0    1    0             4            55          7          9       0       0   
1    0    0            19            23          9          4       0       0   
2    1    0           235            56          3          8       0       0   
3    0    1            21            41          7          7       0       0   
4    2    0           109            92          8         11       0       0   

   HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  Journey  \
0        0        0        0     5     7        2           -2       20   
1        0        0        0    12    -3        9            5       20   
2        1        0        0    -6    -4       -2           -5       20   
3        0        0        0     9     9       -5            0       20   
4        0        0        0     6    -2        9           -3       20   

   Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  Med_ATP  \
0      0.25      0.35         0.10            -0.10     1.70      1.6   
1      0.60     -0.15         0.45             0.25     1.75      1.3   
2     -0.30     -0.20        -0.10            -0.25     0.90      1.0   
3      0.45      0.45        -0.25             0.00     1.45      1.7   
4      0.30     -0.10         0.45            -0.15     1.65      1.2   

   Med_HTGS  Med_ATGS  Med_HTGC  Med_ATGC  year  month  day  
0      1.10      1.00      0.85      0.65  2022     12   12  
1      1.10      1.05      0.50      1.20  2022     12   11  
2      0.45      1.05      0.75      1.25  2022     12   11  
3      1.15      0.90      0.70      0.45  2022     12   11  
4      1.15      0.60      0.85      0.70  2022     12   11

In [27]:
model_fin = CatBoostClassifier(random_state = 0, silent = True, thread_count = 1, iterations = 1100, learning_rate = 0.02, 
                           max_depth = 5).fit(X, y)

y_pred = model_fin.predict(testeo_encode)
y_pred

array([[2],
       [2],
       [1],
       [1],
       [2],
       [2],
       [1],
       [2],
       [1],
       [2],
       [2]], dtype=int64)

Alavés - Levante: 1DNV 1.73</font>

<font color='red'>Las Palmas - Albacete: 1 1.75</font>

<font color='green'>Racing Sant - CD Mirandés: X2 1.67</font>

<font color='blue'>Granada - Burgos: 2Handicap +1.0 1.73</font>

<font color='green'>Eibar - Oviedo: 1 1.85</font>

<font color='green'>Ponferradina - Lugo: 1DNV 1.57</font>

<font color='green'>Sporting Gijón - Cartagena: X2 1.80</font>

<font color='blue'>UD Ibiza - Málaga: 1DNV 1.62</font>

<font color='red'>Zaragoza - Huesca: X2 1.83</font>

<font color='blue'>Andorra - Leganés: 1DNV 1.57</font>

<font color='green'>Villarreal B - Tenerife: 1X 1.50</font>

In [56]:
testeo

Id  Div        HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD  \
0    0  SP2          Alaves      Levante   2.37    3.0   3.25  2.40  2.95   
1    1  SP2      Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25   
2    2  SP2     Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95   
3    3  SP2         Granada       Burgos   1.80    3.3   5.00  1.80  3.20   
4    4  SP2           Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20   
5    5  SP2    Ponferradina         Lugo   2.25    3.3   3.25  2.20  3.20   
6    6  SP2  Sporting Gijon    Cartagena   2.20    3.1   3.50  2.10  3.10   
7    7  SP2        UD Ibiza       Malaga   2.30    3.0   3.40  2.25  2.95   
8    8  SP2        Zaragoza       Huesca   1.95    3.0   4.75  1.95  3.00   
9    9  SP2         Andorra      Leganes   2.10    3.1   3.80  2.05  3.95   
10  10  SP2    Villarreal B     Tenerife   2.90    3.0   2.62  2.90  2.87   

     BWA  Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3  \
0   2.90    2223    22    20    17    13   34   32   W   W   D   D   W   D   
1   4.60    2223    22    21    10    24   35   26   D   L   D   L   W   W   
2   3.50    2223     9    21    15    25   18   20   W   D   L   W   L   L   
3   4.40    2223    23    18    14     9   29   34   L   L   W   W   L   D   
4   4.75    2223    23    12    17    14   33   24   D   W   D   D   L   W   
5   3.00    2223    18    17    27    25   18   18   L   D   D   L   L   D   
6   3.40    2223    22    23    21    18   26   31   D   D   D   L   D   W   
7   3.20    2223    13    14    27    24   15   15   D   D   L   D   L   W   
8   3.90    2223    14    16    17    14   22   27   L   W   D   L   D   D   
9   3.60    2223    20    19    15    18   27   26   L   W   W   D   W   W   
10  2.45    2223    21    19    22    21   27   23   L   W   W   W   L   L   

   HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  \
0    L   W   L   D        WDWLL        WDDWD          7          9       0   
1    W   L   D   D        DDWWD        LLWLD          9          4       0   
2    L   W   L   D        WLLLL        DWLWD          3          8       0   
3    W   W   D   L        LWLWD        LWDWL          7          7       0   
4    W   W   W   D        DDLWW        WDWWD          8         11       0   
5    L   W   D   L        LDLLD        DLDWL          2          5       0   
6    L   W   W   L        DDDLW        DLWWL          6          7       0   
7    L   L   W   D        DLLLW        DDWLD          4          6       0   
8    W   L   D   W        LDDWD        WLDLW          6          7       0   
9    L   D   L   D        LWWLL        WDWDD          6          9       0   
10   D   L   W   W        LWLDW        WWLLW          7          9       0   

    HTWin5  HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  \
0        0        0        0        0     5     7        2           -2   
1        0        0        0        0    12    -3        9            5   
2        0        1        0        0    -6    -4       -2           -5   
3        0        0        0        0     9     9       -5            0   
4        0        0        0        0     6    -2        9           -3   
5        0        0        0        0    -9    -8        0           -3   
6        0        0        0        0     1     5       -5           -1   
7        0        0        0        0   -14   -10        0           -2   
8        0        0        0        0    -3     2       -5           -1   
9        0        0        0        0     5     1        1           -3   
10       0        0        0        0    -1    -2        4           -2   

    Journey  Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  \
0        20      0.25      0.35         0.10            -0.10     1.70   
1        20      0.60     -0.15         0.45             0.25     1.75   
2        20     -0.30     -0.20        -0.10            -0.25     0.90   
3       

In [57]:
testeo['Prediccion'] = y_pred
testeo

Id  Div        HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD  \
0    0  SP2          Alaves      Levante   2.37    3.0   3.25  2.40  2.95   
1    1  SP2      Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25   
2    2  SP2     Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95   
3    3  SP2         Granada       Burgos   1.80    3.3   5.00  1.80  3.20   
4    4  SP2           Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20   
5    5  SP2    Ponferradina         Lugo   2.25    3.3   3.25  2.20  3.20   
6    6  SP2  Sporting Gijon    Cartagena   2.20    3.1   3.50  2.10  3.10   
7    7  SP2        UD Ibiza       Malaga   2.30    3.0   3.40  2.25  2.95   
8    8  SP2        Zaragoza       Huesca   1.95    3.0   4.75  1.95  3.00   
9    9  SP2         Andorra      Leganes   2.10    3.1   3.80  2.05  3.95   
10  10  SP2    Villarreal B     Tenerife   2.90    3.0   2.62  2.90  2.87   

     BWA  Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3  \
0   2.90    2223    22    20    17    13   34   32   W   W   D   D   W   D   
1   4.60    2223    22    21    10    24   35   26   D   L   D   L   W   W   
2   3.50    2223     9    21    15    25   18   20   W   D   L   W   L   L   
3   4.40    2223    23    18    14     9   29   34   L   L   W   W   L   D   
4   4.75    2223    23    12    17    14   33   24   D   W   D   D   L   W   
5   3.00    2223    18    17    27    25   18   18   L   D   D   L   L   D   
6   3.40    2223    22    23    21    18   26   31   D   D   D   L   D   W   
7   3.20    2223    13    14    27    24   15   15   D   D   L   D   L   W   
8   3.90    2223    14    16    17    14   22   27   L   W   D   L   D   D   
9   3.60    2223    20    19    15    18   27   26   L   W   W   D   W   W   
10  2.45    2223    21    19    22    21   27   23   L   W   W   W   L   L   

   HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  \
0    L   W   L   D        WDWLL        WDDWD          7          9       0   
1    W   L   D   D        DDWWD        LLWLD          9          4       0   
2    L   W   L   D        WLLLL        DWLWD          3          8       0   
3    W   W   D   L        LWLWD        LWDWL          7          7       0   
4    W   W   W   D        DDLWW        WDWWD          8         11       0   
5    L   W   D   L        LDLLD        DLDWL          2          5       0   
6    L   W   W   L        DDDLW        DLWWL          6          7       0   
7    L   L   W   D        DLLLW        DDWLD          4          6       0   
8    W   L   D   W        LDDWD        WLDLW          6          7       0   
9    L   D   L   D        LWWLL        WDWDD          6          9       0   
10   D   L   W   W        LWLDW        WWLLW          7          9       0   

    HTWin5  HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  \
0        0        0        0        0     5     7        2           -2   
1        0        0        0        0    12    -3        9            5   
2        0        1        0        0    -6    -4       -2           -5   
3        0        0        0        0     9     9       -5            0   
4        0        0        0        0     6    -2        9           -3   
5        0        0        0        0    -9    -8        0           -3   
6        0        0        0        0     1     5       -5           -1   
7        0        0        0        0   -14   -10        0           -2   
8        0        0        0        0    -3     2       -5           -1   
9        0        0        0        0     5     1        1           -3   
10       0        0        0        0    -1    -2        4           -2   

    Journey  Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  \
0        20      0.25      0.35         0.10            -0.10     1.70   
1        20      0.60     -0.15         0.45             0.25     1.75   
2        20     -0.30     -0.20        -0.10            -0.25     0.90   
3       

In [58]:
res_dict_fin ={
    0: 'Draw',
    1: 'Away',
    2: 'Home'
}
testeo['Prediccion'] = testeo['Prediccion'].replace(res_dict_fin)
testeo

Id  Div        HomeTeam     AwayTeam  B365H  B365D  B365A   BWH   BWD  \
0    0  SP2          Alaves      Levante   2.37    3.0   3.25  2.40  2.95   
1    1  SP2      Las Palmas     Albacete   1.80    3.3   5.00  1.75  3.25   
2    2  SP2     Racing Sant  CD Mirandes   2.10    3.1   4.00  2.10  2.95   
3    3  SP2         Granada       Burgos   1.80    3.3   5.00  1.80  3.20   
4    4  SP2           Eibar       Oviedo   1.75    3.4   4.75  1.75  3.20   
5    5  SP2    Ponferradina         Lugo   2.25    3.3   3.25  2.20  3.20   
6    6  SP2  Sporting Gijon    Cartagena   2.20    3.1   3.50  2.10  3.10   
7    7  SP2        UD Ibiza       Malaga   2.30    3.0   3.40  2.25  2.95   
8    8  SP2        Zaragoza       Huesca   1.95    3.0   4.75  1.95  3.00   
9    9  SP2         Andorra      Leganes   2.10    3.1   3.80  2.05  3.95   
10  10  SP2    Villarreal B     Tenerife   2.90    3.0   2.62  2.90  2.87   

     BWA  Season  HTGS  ATGS  HTGC  ATGC  HTP  ATP HM1 AM1 HM2 AM2 HM3 AM3  \
0   2.90    2223    22    20    17    13   34   32   W   W   D   D   W   D   
1   4.60    2223    22    21    10    24   35   26   D   L   D   L   W   W   
2   3.50    2223     9    21    15    25   18   20   W   D   L   W   L   L   
3   4.40    2223    23    18    14     9   29   34   L   L   W   W   L   D   
4   4.75    2223    23    12    17    14   33   24   D   W   D   D   L   W   
5   3.00    2223    18    17    27    25   18   18   L   D   D   L   L   D   
6   3.40    2223    22    23    21    18   26   31   D   D   D   L   D   W   
7   3.20    2223    13    14    27    24   15   15   D   D   L   D   L   W   
8   3.90    2223    14    16    17    14   22   27   L   W   D   L   D   D   
9   3.60    2223    20    19    15    18   27   26   L   W   W   D   W   W   
10  2.45    2223    21    19    22    21   27   23   L   W   W   W   L   L   

   HM4 AM4 HM5 AM5 HTFormPtsStr ATFormPtsStr  HTFormPts  ATFormPts  HTWin3  \
0    L   W   L   D        WDWLL        WDDWD          7          9       0   
1    W   L   D   D        DDWWD        LLWLD          9          4       0   
2    L   W   L   D        WLLLL        DWLWD          3          8       0   
3    W   W   D   L        LWLWD        LWDWL          7          7       0   
4    W   W   W   D        DDLWW        WDWWD          8         11       0   
5    L   W   D   L        LDLLD        DLDWL          2          5       0   
6    L   W   W   L        DDDLW        DLWWL          6          7       0   
7    L   L   W   D        DLLLW        DDWLD          4          6       0   
8    W   L   D   W        LDDWD        WLDLW          6          7       0   
9    L   D   L   D        LWWLL        WDWDD          6          9       0   
10   D   L   W   W        LWLDW        WWLLW          7          9       0   

    HTWin5  HTLoss3  HTLoss5  ATLoss3  HTDG  ATDG  DiffPts  DiffFormPts  \
0        0        0        0        0     5     7        2           -2   
1        0        0        0        0    12    -3        9            5   
2        0        1        0        0    -6    -4       -2           -5   
3        0        0        0        0     9     9       -5            0   
4        0        0        0        0     6    -2        9           -3   
5        0        0        0        0    -9    -8        0           -3   
6        0        0        0        0     1     5       -5           -1   
7        0        0        0        0   -14   -10        0           -2   
8        0        0        0        0    -3     2       -5           -1   
9        0        0        0        0     5     1        1           -3   
10       0        0        0        0    -1    -2        4           -2   

    Journey  Med_HTDG  Med_ATDG  Med_DiffPts  Med_DiffFormPts  Med_HTP  \
0        20      0.25      0.35         0.10            -0.10     1.70   
1        20      0.60     -0.15         0.45             0.25     1.75   
2        20     -0.30     -0.20        -0.10            -0.25     0.90   
3       

In [59]:
testeo.to_csv("../predicciones.csv", index = False)